src/internal/analytics/interfaces.ts

```diff
@@ -182,12 +182,25 @@ export interface IPerformanceMetrics {
   taskCompletionData: TaskCompletionDataMethod;
 }
 
+type JSONValue = string | number | boolean | null | undefined;
```

Comment:
When the value of an object's property is `undefined`, the property will not appear in the JSONified string at all - is that okay for our metrics? Should we enforce values being `null` instead?

Reply:
I had mostly tried to align with this [type interface](https://github.com/cloudscape-design/component-toolkit/blob/8ef08a77a38f6c49a2c30751f1fe01f4124345de/src/internal/base-component/metrics/interfaces.ts#L4). I think not appearing in the string is fine.